src/loaders/index.ts

```diff
@@ -0,0 +1,46 @@
+import { LoadSpec, loadSpecToString, IVolumeLoader, VolumeDims } from "./IVolumeLoader";
+
+import { OMEZarrLoader } from "./OmeZarrLoader";
+import { JsonImageInfoLoader } from "./JsonImageInfoLoader";
+import { TiffLoader } from "./TiffLoader";
+import VolumeCache from "../VolumeCache";
+
+const enum VolumeFileFormat {
+  ZARR = "zarr",
+  JSON = "json",
+  TIFF = "tiff",
+}
+
+type CreateLoaderOptions = {
+  path: string | string[];
+  cache?: VolumeCache;
+  scene?: number;
+};
+
+const forceString = (value: string | string[]): string => (typeof value === "object" ? value[0] : value);
+
+async function createVolumeLoader(fileType: VolumeFileFormat, options: CreateLoaderOptions): Promise<IVolumeLoader> {
+  switch (fileType) {
+    case VolumeFileFormat.ZARR:
+      return await OMEZarrLoader.createLoader(forceString(options.path), options.scene, options.cache);
+    case VolumeFileFormat.JSON:
+      return new JsonImageInfoLoader(options.path, options.cache);
+    case VolumeFileFormat.TIFF:
```

Comment:
I have a vague suspicion that w3cv is going to have a kind of switch statement that guesses the VolumeFileFormat from the file extension, which might feel redundant with this code.   Not sure what would be cleanest, but maybe the type could be used as a hint and if omitted, then we make a guess by looking at the end of the string, or if it's an array.  Then fileType could become an optional member of CreateLoaderOptions, even.